<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>Burns Life Sciences Consulting GmbH</title>
<meta name="generator" content="Org Mode" />
<link rel="stylesheet" href="simple.css" />
<style>
  .dark-mode {
    --bg: #212121;
    --accent-bg: #2b2b2b;
    --text: #dcdcdc;
    --text-light: #ababab;
    --accent: #ffb300;
    /*--code: #f06292;*/
    --preformatted: #ccc;
    --disabled: #111;
  }
  .dark-mode img,
  .dark-mode video {
    opacity: 0.8;
  }
  .toggle-theme-btn {
      transform: scale(0.7);
      opacity: 0.4;
      transition: opacity 0.2s;
    }
    .toggle-theme-btn:hover {
      opacity: 0.4;
    }
</style>
<script>
 function toggleDarkMode() {
    const body = document.body;
    body.classList.toggle("dark-mode");
    // Save the user's theme preference to localStorage
    if (body.classList.contains("dark-mode")) {
      localStorage.setItem("theme", "dark");
    } else {
      localStorage.setItem("theme", "light");
    }
  }

  function setDefaultDarkMode() {
    // Retrieve the user's theme preference from localStorage
    const storedTheme = localStorage.getItem("theme");

    // If the stored theme is light, do nothing; otherwise, set it to dark
    if (storedTheme !== "light") {
      document.body.classList.add("dark-mode");
    }
  }

  // Set the default mode to dark when the DOM is fully loaded
  document.addEventListener('DOMContentLoaded', setDefaultDarkMode);
</script>
</head>
<body>
<div id="preamble" class="status">
<div style="position: fixed; top: 10px; right: 10px;">
         <button class="toggle-theme-btn" onclick="toggleDarkMode()">Light/Dark</button>
       </div>
</div>
<div id="content" class="content">
<h1 class="title">Burns Life Sciences Consulting GmbH</h1>
<div id="top"></div>

<p>
For Tyler's personal website, go <a href="./index.html">here</a>.
</p>

<div id="table-of-contents" role="doc-toc">
<h2>Table of Contents</h2>
<div id="text-table-of-contents" role="doc-toc">
<ul>
<li><a href="#orgfaa3741">Our philosophy</a></li>
<li><a href="#org9a4bdac">Who are we</a></li>
<li><a href="#org53f931d">Who we help</a></li>
<li><a href="#org2a00562">Services</a></li>
<li><a href="#org5c4a089">Domains</a></li>
<li><a href="#org6fede40">What to expect from an engagement with Burns LSC</a></li>
<li><a href="#org0351bd9">What people are saying</a></li>
<li><a href="#orgac45f3c">Contact us</a></li>
<li><a href="#org88a9f83">Insights</a></li>
</ul>
</div>
</div>

<div id="outline-container-orgd8b4d49" class="outline-2">
<h2 id="orgd8b4d49">Your single-cell and spatial data: clear, actionable, now</h2>
<div class="outline-text-2" id="text-orgd8b4d49">
<p>
Struggling with slow, convoluted single-cell or spatial data analyses? We provide rapid, next-level insights that transform months-long projects into an efficient pipeline—so you can make decisions fast.
</p>

<p>
We’ve been leading the charge in single-cell and spatial analysis since the early 2010s, pioneering CyTOF and adopting AI tools (like GPT-3, early access) well before they went mainstream. Our approach isn’t just about handing over a fancy report; we teach your team to replicate and refine the process, saving you hundreds of thousands in future service fees.
</p>

<p>
Here’s what sets us apart:
</p>

<ol class="org-ol">
<li>Fast-Track Analysis – We compress your research timelines significantly, going from data to actionable insights in a fraction of the usual time.</li>
<li>Hands-On Training – Your team learns our streamlined methodology, so you remain self-sufficient long after we wrap up.</li>
<li>Future-Proof Expertise – Stay on the cutting edge with our battle-tested AI strategies, honed before ChatGPT took the world by storm.</li>
</ol>

<p>
If you want to accelerate your breakthroughs and empower your team with the latest in single-cell and spatial analytics, reach out now. We only accept a limited number of new clients each quarter, so don’t miss your chance. Let’s make your data work for you—faster and smarter.
</p>
</div>
</div>
<div id="outline-container-orgfaa3741" class="outline-2">
<h2 id="orgfaa3741">Our philosophy</h2>
<div class="outline-text-2" id="text-orgfaa3741">
<a href="#top">Back to top</a>

<div id="orgfd17e1a" class="figure">
<p><img src="images/2024-09-08_13-45-17_matrix_multiplication.jpg" alt="2024-09-08_13-45-17_matrix_multiplication.jpg" />
</p>
</div>

<p>
Biological datasets are becoming increasingly complicated, as are the relevant computational tools to process and analyze the datasets.
</p>

<p>
As such, we maintain a "first principles" approach to every project, drilling down on the core concepts that do not change, in the rapidly changing environment that we have all become used to. This approach allows us to seamlessly integrate the latest tools, such as AI, into our workflows while remaining grounded in timeless principles.
</p>

<p>
Whatever the project, we will cut through the noise and get right to the essential elements needed to deliver the highest quality and highest value solutions. And we will show your team how to do it.
</p>
</div>
</div>
<div id="outline-container-org9a4bdac" class="outline-2">
<h2 id="org9a4bdac">Who are we</h2>
<div class="outline-text-2" id="text-org9a4bdac">
<a href="#top">Back to top</a>
</div>
<div id="outline-container-orgb3ea9b5" class="outline-3">
<h3 id="orgb3ea9b5">Tyler Burns</h3>
<div class="outline-text-3" id="text-orgb3ea9b5">

<div id="org5d5528d" class="figure">
<p><img src="images/2024-09-08_13-52-02_professional_photo.jpg" alt="2024-09-08_13-52-02_professional_photo.jpg" width="40%" style="float:left;margin:0px 20px 20px 0px;" />
</p>
</div>

<p>
Tyler has a PhD in Cancer Biology from the laboratory of Garry Nolan at Stanford University. Starting out as a wet-lab biologist, he developed a passion for software engineering and data science in graduate school, and is now a hybrid biologist, leveraging both domains to solve complex problems.
</p>

<p>
His vision: by using single-cell technologies, we develop and leverage a deeper understanding of heterogeneity, in order to treat and cure diseases often called heterogeneous (eg. cancer).
</p>

<p>
He has been doing consulting for industry since his final year in graduate school in 2016, with single-cell and spatial domain expertise dating back to 2012. His work across various companies in various domains gives him a powerful outsider perspective to get you what you need, right now.
</p>

<p>
To see Tyler's personal website, complete with blog posts, personal projects, and samples of work, please go <a href="https://tjburns08.github.io/">here</a>.
</p>
</div>
</div>
<div id="outline-container-orgaaf9caa" class="outline-3">
<h3 id="orgaaf9caa">The Burns LSC Network</h3>
<div class="outline-text-3" id="text-orgaaf9caa">
<p>
Burns Life Sciences Consulting operates as a decentralized network of top talent in the single cell and spatial fields, amassed over the 12+ years of doing this work, starting with Tyler's work in the lab of Garry Nolan in 2012 when CyTOF and its analysis were still in the "wild west" phase. We have deep experts in each of the domains below at our fingertips.
</p>
</div>
</div>
</div>
<div id="outline-container-org53f931d" class="outline-2">
<h2 id="org53f931d">Who we help</h2>
<div class="outline-text-2" id="text-org53f931d">
<a href="#top">Back to top</a>
<ul class="org-ul">
<li>Companies developing single-cell and spatial technologies</li>
<li>Companies using single-cell and spatial technologies (eg. pharma)</li>
<li>PIs doing single-cell and spatial research</li>
<li>(for free) PhD students and postdocs who may eventually be in the three buckets above</li>
</ul>
</div>
</div>
<div id="outline-container-org2a00562" class="outline-2">
<h2 id="org2a00562">Services</h2>
<div class="outline-text-2" id="text-org2a00562">
<a href="#top">Back to top</a>

<div id="org866a5b3" class="figure">
<p><img src="images/2024-09-08_13-45-49_consultant_laptop.jpg" alt="2024-09-08_13-45-49_consultant_laptop.jpg" />
</p>
</div>
</div>
<div id="outline-container-org485fe70" class="outline-3">
<h3 id="org485fe70">Data analysis</h3>
<div class="outline-text-3" id="text-org485fe70">
<p>
Struggling with slow, convoluted pipelines and issues like batch effects? We build and use custom-tailored data analysis pipelines for each of the listed domains. We focus heavily on common pain points (eg. batch effects, cell segmentation) so you don't have to.
</p>

<p>
Specifically, we can:
</p>
<ul class="org-ul">
<li>Analyze Your Data: Hand it over and let us handle the heavy lifting.</li>
<li>Build and Transfer Tools: Need bespoke R or Python packages? We’ll develop them for you to keep.</li>
</ul>

<p>
We have the following options:
</p>
<ul class="org-ul">
<li>Done for you: you do nothing</li>
<li>Done with you: we provide a helping hand</li>
<li>Advisory: we guide your data analysis journey</li>
</ul>

<p>
Regardless of the option above, we will show you exactly how it's done, free of charge.
</p>

<p>
Result: You’ll make quicker decisions and stay ahead in the “big data” + AI landscape, without having to reinvent the wheel.
</p>
</div>
</div>
<div id="outline-container-org66b6651" class="outline-3">
<h3 id="org66b6651">Teaching</h3>
<div class="outline-text-3" id="text-org66b6651">
</div>
<div id="outline-container-org91dfa43" class="outline-4">
<h4 id="org91dfa43">Introduction to single-cell bioinformatics seminar</h4>
<div class="outline-text-4" id="text-org91dfa43">
<p>
Accelerate your team’s proficiency in single-cell analysis with our intensive seminar—delivered in person or online. Over three days, we tackle coding fundamentals, explore popular libraries, and practice real-world analyses.
</p>

<ul class="org-ul">
<li>Core Curriculum: Understand the "why" behind single-cell bioinformatics instead of treating it like a black box.</li>
<li>Hands-On Practice: Write simple programs, master the basics of R and Python scripts, and learn to apply no-code tools effectively (because you'll understand them).</li>
<li>Follow-Up Support: We remain available to help apply newfound skills directly to your current research projects.</li>
</ul>

<p>
Result: A confident, self-sufficient team that can manage and interpret single-cell data—from foundational coding to complex analyses.
</p>
</div>
</div>
</div>
</div>
<div id="outline-container-org5c4a089" class="outline-2">
<h2 id="org5c4a089">Domains</h2>
<div class="outline-text-2" id="text-org5c4a089">
<a href="#top">Back to top</a>

<div id="orgda883ee" class="figure">
<p><img src="images/2024-09-08_13-46-48_tubes.jpg" alt="2024-09-08_13-46-48_tubes.jpg" />
</p>
</div>

<p>
The following are brief descriptions of the domains in which we have deep expertise, and through which we provide our services.
</p>
</div>
<div id="outline-container-orgda1b93a" class="outline-3">
<h3 id="orgda1b93a">High-dimensional flow cytometry</h3>
<div class="outline-text-3" id="text-orgda1b93a">
</div>
<div id="outline-container-org446254a" class="outline-4">
<h4 id="org446254a">Mass cytometry</h4>
<div class="outline-text-4" id="text-org446254a">
<p>
We leverage direct wet-lab and dry-lab experience from the lab of Garry Nolan starting in 2012, when mass cytometry was in its infancy, and the best practices were being developed in real time.
</p>
</div>
</div>
<div id="outline-container-org030072c" class="outline-4">
<h4 id="org030072c">Spectral flow cytometry</h4>
<div class="outline-text-4" id="text-org030072c">
<p>
Leveraging our experience in mass cytometry and traditional fluorescnece flow cytometry, we began working and partnering in spectral in the 2020s.
</p>
</div>
</div>
</div>
<div id="outline-container-org10df823" class="outline-3">
<h3 id="org10df823">Single-cell sequencing</h3>
<div class="outline-text-3" id="text-org10df823">
</div>
<div id="outline-container-org29288de" class="outline-4">
<h4 id="org29288de">RNA, ATAC, and related (integration included)</h4>
<div class="outline-text-4" id="text-org29288de">
<p>
Leveraging established best practices and our "first principles" approach from the early days of mass cytometry, we make sure that we deeply understand every detail of our analysis solutions. Our services here include data integration.
</p>
</div>
</div>
<div id="outline-container-org9b0f134" class="outline-4">
<h4 id="org9b0f134">CITE-seq and related</h4>
<div class="outline-text-4" id="text-org9b0f134">
<p>
Here, we leverage our work in flow and mass cytometry to make sure the antibody side is done right, and looks right, along with our expertise in single-cell sequencing on the transcript side.
</p>
</div>
</div>
</div>
<div id="outline-container-org493e353" class="outline-3">
<h3 id="org493e353">Spatial proteomics</h3>
<div class="outline-text-3" id="text-org493e353">
</div>
<div id="outline-container-org30ff34f" class="outline-4">
<h4 id="org30ff34f">Mass tag based (MIBI and IMC), and fluorescence based (codex)</h4>
<div class="outline-text-4" id="text-org30ff34f">
<p>
We utilize our direct experience from the Nolan lab and collaborators, where both MIBI and CODEX were being developed at the same time.
</p>

<p>
This is how we deliver the same "first principles" approach that we deliver for our suspension-based analyses. We emphasize getting all the painful details correct, like cell segmentation.
</p>
</div>
</div>
</div>
<div id="outline-container-org34d3521" class="outline-3">
<h3 id="org34d3521">Spatial transcriptomics</h3>
<div class="outline-text-3" id="text-org34d3521">
</div>
<div id="outline-container-orgee43db4" class="outline-4">
<h4 id="orgee43db4">Core based (eg. Visium), and cell based (eg. Xenium)</h4>
<div class="outline-text-4" id="text-orgee43db4">
<p>
These methods share principles and methods with spatial proteomic analyis. Again, cell segmentation is critical. There are key differences, and we note that there are a number of different technologies to choose from.
</p>

<p>
We emphasize a general understanding of each one and the respective trade-offs, such that we can help you navigate this field as it develops and give you what you need relevant to your project.
</p>
</div>
</div>
</div>
<div id="outline-container-orgcc95e34" class="outline-3">
<h3 id="orgcc95e34">Network analysis</h3>
<div class="outline-text-3" id="text-orgcc95e34">
</div>
<div id="outline-container-orgeb67383" class="outline-4">
<h4 id="orgeb67383">Pathway analysis</h4>
<div class="outline-text-4" id="text-orgeb67383">
<p>
We determine what pathways are likely active in a given cell subset with specific DEGs. We then go into the pathways themselves and determine what genes within the pathway matter. We emphasize the fundamentals of graph theory in this type of work.
</p>
</div>
</div>
</div>
<div id="outline-container-org052f9ae" class="outline-3">
<h3 id="org052f9ae">Data management</h3>
<div class="outline-text-3" id="text-org052f9ae">
</div>
<div id="outline-container-orgff04899" class="outline-4">
<h4 id="orgff04899">Knowledge graphs (eg. Neo4J)</h4>
<div class="outline-text-4" id="text-orgff04899">
<p>
Projects in this domain typically involve organizations that have multi-modal data (genes, drugs, pathways, PPI, diseases, and so forth). Knowledge graphs can put all of this together to produce novel insights you won't get looking at these data separately.
</p>
</div>
</div>
<div id="outline-container-org43932cb" class="outline-4">
<h4 id="org43932cb">LLM integration</h4>
<div class="outline-text-4" id="text-org43932cb">
<p>
A typical project in this domain includes converting English into a given query language (eg. Cypher for Neo4J), such that a knowledge graph can be easily queried to deliver actionable insights.
</p>
</div>
</div>
</div>
</div>
<div id="outline-container-org6fede40" class="outline-2">
<h2 id="org6fede40">What to expect from an engagement with Burns LSC</h2>
<div class="outline-text-2" id="text-org6fede40">
<a href="#top">Back to top</a>

<div id="orgb87199c" class="figure">
<p><img src="images/2024-09-08_15-25-11_handshake.jpg" alt="2024-09-08_15-25-11_handshake.jpg" />
</p>
</div>

<p>
A typical engagement begins with a mutual nondisclosure agreement (NDA), so we can both talk freely. The project can either be fixed-scope (eg. a specific thing needs to be analyzed by a particular time), or open-ended (eg. continuous advisory). When the details, scope, and deliverables are decided upon, then a statement of work is finalized (sometimes from Burns, sometimes from the client) and signed.
</p>

<p>
The project proceeds with a minimum of one video call per week, so we can always be sync'd on progress toward milestones, and any new developments on both sides. Any adjustments to the contract after it has been signed can be done in writing.
</p>

<p>
Although most of our client interactions happen remotely (via email, Slack, and video calls), we strongly recommend one or more on-site visits per year—particularly for longer-term engagements. In our experience, meeting in person significantly strengthens the consultant–client relationship and fosters deeper collaboration.
</p>
</div>
</div>
<div id="outline-container-org0351bd9" class="outline-2">
<h2 id="org0351bd9">What people are saying</h2>
<div class="outline-text-2" id="text-org0351bd9">
<a href="#top">Back to top</a>
<p>
<i>“BLS Consulting has brought tremendous value to Alkahest’s research efforts. Their extensive expertise helped us advance cutting-edge projects both in the scientific foundation, as well as the technical implementation. They’re the rare organization that can distill the ambiguity of advanced research down to pragmatic solutions with demonstrable value.”</i>
</p>

<p>
— Scott Lohr, Senior Director of Informatics &amp; Engineering, Alkahest Inc.
</p>

<p>
<i>"Tyler’s expertise in data science, deep biological knowledge, and mastery of cutting-edge technologies have been instrumental in advancing several of our projects. His innovative thinking, coupled with his organization and responsiveness, make him an invaluable consultant. Tyler is our go-to person whenever we encounter challenges, and I greatly appreciate his commitment to efficiency and passion for both the team and the science."</i>
</p>

<p>
— Bahareh Ajami, Assistant Professor of Immunology, Oregon Health &amp; Science University
</p>

<p>
<i>“Tyler’s sharp data science expertise on top of his deep biological background and first principles mindset has given us quality solutions every time we’ve brought him in.”</i>
</p>

<p>
— Xiaoyan Qian, Senior Computational Biologist, 10x Genomics
</p>

<p>
<i>“Tyler’s innovative data analysis solutions has transformed our raw data into actionable insights, enabling us to make smarter and more informed decisions going forward. His fast and engaged approach makes working with him very enjoyable and informative, we look forward to engaging his services again in future projects!”</i>
</p>

<p>
— Natalia Becares, Principal Scientist, LIfT Biosciecnes
</p>
</div>
</div>
<div id="outline-container-orgac45f3c" class="outline-2">
<h2 id="orgac45f3c">Contact us</h2>
<div class="outline-text-2" id="text-orgac45f3c">
<a href="#top">Back to top</a>
</div>
<div id="outline-container-orgb82b684" class="outline-3">
<h3 id="orgb82b684"><a href="https://calendly.com/burnslsc-info/30min">Book a free 30min consultation</a></h3>
<div class="outline-text-3" id="text-orgb82b684">
<p>
If you'd like my two cents on anything, and/or you would like to explore potential engagements/partnerships, you can book a call accordingly.
</p>
</div>
</div>
<div id="outline-container-orge856d29" class="outline-3">
<h3 id="orge856d29"><a href="https://www.linkedin.com/in/tylerjburns/">Tyler's LinkedIn</a></h3>
<div class="outline-text-3" id="text-orge856d29">
<p>
Tyler maintains a strong LinkedIn presence. If you're not connected, simply send a connection request with a message. If he can't answer your questions, he can link you to people who can.
</p>
</div>
</div>
</div>
<div id="outline-container-org88a9f83" class="outline-2">
<h2 id="org88a9f83">Insights</h2>
<div class="outline-text-2" id="text-org88a9f83">
<a href="#top">Back to top</a>
</div>
<div id="outline-container-org9520c1f" class="outline-3">
<h3 id="org9520c1f"><a href="https://tjburns08.github.io/social_media_posts.html">A compendium of Tyler's social media posts</a></h3>
<div class="outline-text-3" id="text-org9520c1f">
<p>
These are grouped by subject, and easier to go through here than trying to browse through LinkedIn history.
</p>
</div>
</div>
<div id="outline-container-orgb1f98af" class="outline-3">
<h3 id="orgb1f98af"><a href="https://tjburns08.github.io/index.html#single_cell">Single-cell relevant posts</a></h3>
<div class="outline-text-3" id="text-orgb1f98af">
<p>
These are markdowns and articles related to all things single-cell, from Tyler's personal website.
</p>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="creator"><a href="https://www.gnu.org/software/emacs/">Emacs</a> 28.1 (<a href="https://orgmode.org">Org</a> mode 9.5.2)</p>
</div>
</body>
</html>
