<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>Burns Life Sciences Consulting GmbH</title>
<meta name="generator" content="Org Mode" />
<style>
  #content { max-width: 60em; margin: auto; }
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #e6e6e6;
    border-radius: 3px;
    background-color: #f2f2f2;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
  }
  pre.src:before {
    display: none;
    position: absolute;
    top: -8px;
    right: 12px;
    padding: 3px;
    color: #555;
    background-color: #f2f2f299;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-authinfo::before { content: 'Authinfo'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
</style>
<link rel="stylesheet" href="simple.css" />
<style>
  .dark-mode {
    --bg: #212121;
    --accent-bg: #2b2b2b;
    --text: #dcdcdc;
    --text-light: #ababab;
    --accent: #ffb300;
    /*--code: #f06292;*/
    --preformatted: #ccc;
    --disabled: #111;
  }
  .dark-mode img,
  .dark-mode video {
    opacity: 0.8;
  }
  .toggle-theme-btn {
      transform: scale(0.7);
      opacity: 0.4;
      transition: opacity 0.2s;
    }
    .toggle-theme-btn:hover {
      opacity: 0.4;
    }
</style>
<script>
 function toggleDarkMode() {
    const body = document.body;
    body.classList.toggle("dark-mode");
    // Save the user's theme preference to localStorage
    if (body.classList.contains("dark-mode")) {
      localStorage.setItem("theme", "dark");
    } else {
      localStorage.setItem("theme", "light");
    }
  }

  function setDefaultDarkMode() {
    // Retrieve the user's theme preference from localStorage
    const storedTheme = localStorage.getItem("theme");

    // If the stored theme is light, do nothing; otherwise, set it to dark
    if (storedTheme !== "light") {
      document.body.classList.add("dark-mode");
    }
  }

  // Set the default mode to dark when the DOM is fully loaded
  document.addEventListener('DOMContentLoaded', setDefaultDarkMode);
</script>
<script>
// @license magnet:?xt=urn:btih:1f739d935676111cfff4b4693e3816e664797050&amp;dn=gpl-3.0.txt GPL-v3-or-Later
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
// @license-end
</script>
</head>
<body>
<div id="preamble" class="status">
<div style="position: fixed; top: 10px; right: 10px;">
         <button class="toggle-theme-btn" onclick="toggleDarkMode()">Light/Dark</button>
       </div>
</div>
<div id="content" class="content">
<h1 class="title">Burns Life Sciences Consulting GmbH</h1>
<div id="top"></div>

<p>
For Tyler's personal website, go <a href="./index.html">here</a>.
</p>

<div id="table-of-contents" role="doc-toc">
<h2>Table of Contents</h2>
<div id="text-table-of-contents" role="doc-toc">
<ul>
<li><a href="#org19e7dcd">Our philosophy</a></li>
<li><a href="#org58ec366">Who are we</a></li>
<li><a href="#org91cdb72">Who we help</a></li>
<li><a href="#orgcd923f0">Services</a></li>
<li><a href="#org03bbd20">Domains</a></li>
<li><a href="#orgd4894d9">What to expect from an engagement with Burns LSC</a></li>
<li><a href="#org55c671f">What people are saying</a></li>
<li><a href="#org7546634">Contact us</a></li>
<li><a href="#orgcc5f50d">Insights</a></li>
</ul>
</div>
</div>

<div id="outline-container-orgdb63cd0" class="outline-2">
<h2 id="orgdb63cd0">Your single-cell and spatial strategic partner</h2>
<div class="outline-text-2" id="text-orgdb63cd0">
<p>
A partnership with us adds substantial value to all aspects of your single-cell and spatial operations, from the product to the research to the marketing.
</p>
</div>
</div>

<div id="outline-container-org19e7dcd" class="outline-2">
<h2 id="org19e7dcd">Our philosophy</h2>
<div class="outline-text-2" id="text-org19e7dcd">
<a href="#top">Back to top</a>

<div id="orgfcff205" class="figure">
<p><img src="images/2024-09-08_13-45-17_matrix_multiplication.jpg" alt="2024-09-08_13-45-17_matrix_multiplication.jpg" />
</p>
</div>

<p>
Biological datasets are becoming increasingly complicated, as are the relevant computational tools to process and analyze the datasets.
</p>

<p>
As such, we maintain a "first principles" approach to every project, drilling down on the core concepts that do not change, in the rapidly changing environment that we have all become used to. This approach allows us to seamlessly integrate the latest tools, such as AI, into our workflows while remaining grounded in fundamental principles.
</p>

<p>
Whatever the project, we will cut through the noise and get right to the essential elements needed to deliver the highest quality and highest value solutions.
</p>
</div>
</div>
<div id="outline-container-org58ec366" class="outline-2">
<h2 id="org58ec366">Who are we</h2>
<div class="outline-text-2" id="text-org58ec366">
<a href="#top">Back to top</a>
</div>
<div id="outline-container-org48bc029" class="outline-3">
<h3 id="org48bc029">Tyler Burns</h3>
<div class="outline-text-3" id="text-org48bc029">

<div id="orgb50b41e" class="figure">
<p><img src="images/2024-09-08_13-52-02_professional_photo.jpg" alt="2024-09-08_13-52-02_professional_photo.jpg" width="40%" style="float:left;margin:0px 20px 20px 0px;" />
</p>
</div>

<p>
Tyler has a PhD in Cancer Biology from the laboratory of Garry Nolan at Stanford University. Starting out as a wet-lab biologist, he developed a passion for software engineering and data science in graduate school, and is now a hybrid biologist, leveraging both domains to solve complex problems.
</p>

<p>
His vision: by using single-cell technologies, we develop and leverage a deeper understanding of heterogeneity, in order to treat and cure diseases often called heterogeneous (eg. cancer).
</p>

<p>
He has been doing consulting for industry since his final year in graduate school in 2016, with single-cell and spatial domain expertise dating back to 2012. His work across various companies in various domains gives him a powerful outsider perspective to get you what you need, right now.
</p>

<p>
To see Tyler's personal website, complete with blog posts, personal projects, and samples of work, please go <a href="https://tjburns08.github.io/">here</a>.
</p>
</div>
</div>
<div id="outline-container-orged149ca" class="outline-3">
<h3 id="orged149ca">The Burns LSC Network</h3>
<div class="outline-text-3" id="text-orged149ca">
<p>
Burns Life Sciences Consulting operates as a decentralized network of top talent in the single cell and spatial fields, amassed over the 12+ years of doing this work, starting with Tyler's work in the lab of Garry Nolan in 2012 when CyTOF and its analysis were still in the "wild west" phase. We have deep experts in each of the domains below at our fingertips.
</p>
</div>
</div>
</div>
<div id="outline-container-org91cdb72" class="outline-2">
<h2 id="org91cdb72">Who we help</h2>
<div class="outline-text-2" id="text-org91cdb72">
<a href="#top">Back to top</a>
<ul class="org-ul">
<li>Companies developing single-cell and spatial technologies</li>
<li>Companies using single-cell and spatial technologies (eg. pharma)</li>
<li>PIs doing single-cell and spatial research</li>
<li>(for free) PhD students and postdocs who may eventually be in the three buckets above</li>
</ul>
</div>
</div>
<div id="outline-container-orgcd923f0" class="outline-2">
<h2 id="orgcd923f0">Services</h2>
<div class="outline-text-2" id="text-orgcd923f0">
<a href="#top">Back to top</a>

<div id="orgedf05d4" class="figure">
<p><img src="images/2024-09-08_13-45-49_consultant_laptop.jpg" alt="2024-09-08_13-45-49_consultant_laptop.jpg" />
</p>
</div>
</div>
<div id="outline-container-org84b771f" class="outline-3">
<h3 id="org84b771f">Data analysis</h3>
<div class="outline-text-3" id="text-org84b771f">
<p>
We build and use custom-tailored data analysis pipelines for each of the listed domains. We focus heavily on common pain points (eg. batch effects, cell segmentation) so you don't have to.
</p>

<p>
Specifically, we can:
</p>
<ul class="org-ul">
<li>analyze your data</li>
<li>develop and build analysis tools for you to keep (eg. R and python packages)</li>
</ul>

<p>
We have the following options:
</p>
<ul class="org-ul">
<li>Done for you (you do nothing)</li>
<li>Done with you (we provide a helping hand)</li>
<li>Advisory (we guide your data analysis journey)</li>
</ul>

<p>
By working with us, you'll receive actionable insights without having to navigate the increasingly complex "big data" and AI landscapes alone.
</p>
</div>
</div>
<div id="outline-container-orgc8687da" class="outline-3">
<h3 id="orgc8687da">Experimental design</h3>
<div class="outline-text-3" id="text-orgc8687da">
<p>
We understand that bioinformatics is a garbage-in, garbage-out endeavor. As such, we want to make sure that the wet lab procedures are done properly and that the experiment is designed and optimized to best answer the research team's questions.
</p>

<p>
We help you avoid common pitfalls and produce reliable data, which makes all the difference when it comes to the data analysis step.
</p>
</div>
</div>
<div id="outline-container-orga680e83" class="outline-3">
<h3 id="orga680e83">Teaching</h3>
<div class="outline-text-3" id="text-orga680e83">
</div>
<div id="outline-container-org07abeb5" class="outline-4">
<h4 id="org07abeb5">Introduction to single-cell bioinformatics seminar</h4>
<div class="outline-text-4" id="text-org07abeb5">
<p>
This seminar is 3 days. It can be done in person or over video calls. We focus heavily on the fundamentals of how to code, and the basic concepts that underlie the popular packages and libraries being used in modern single cell analysis.
</p>

<p>
The options in terms of delivery are:
</p>
<ul class="org-ul">
<li>I come to you and teach the class in person.</li>
<li>I do the course via videoconferencing (in which case we can extend it past 3 days if you want it to be less intensive).</li>
</ul>

<p>
After the seminar, we offer ongoing support, such that we can help you apply the principles we teach to your projects.
</p>
</div>
</div>
</div>
<div id="outline-container-org089f1c1" class="outline-3">
<h3 id="org089f1c1">Marketing and communication</h3>
<div class="outline-text-3" id="text-org089f1c1">
<p>
For biotech companies, our materials bridge the gap between highly technical information around your products, and the audience of current or potential users that may or may not have the deep domain expertise necessary to make decisions around the use of your products.
</p>

<p>
Marketing copy we produce or help produce includes:
</p>
<ul class="org-ul">
<li>written material (eg. blog posts, manuscripts)</li>
<li>audio/visual material (eg. interviews).</li>
</ul>

<p>
With our help, you will be able to effectively communicate your message to your audience and expand your market reach accordingly.
</p>
</div>
</div>
<div id="outline-container-org5da6bca" class="outline-3">
<h3 id="org5da6bca">Market and technology trends research</h3>
<div class="outline-text-3" id="text-org5da6bca">
<p>
In today's rapidly evolving bio-pharma landscape, staying ahead of emerging trends and technologies is crucial. We provide clear, actionable insights that keep you ahead of your competition and ahead of the curve.
</p>

<p>
Our services include:
</p>
<ul class="org-ul">
<li>Helping organizations prepare for a KOL interview campaign.</li>
<li>Conducting or mediating KOL interviews.</li>
<li>Synthesizing material from many sources (KOLs, media, data, direct experience) into written reports and presentations.</li>
<li>Providing our direct recommendations as to what decisions should be made, given the research that was done.</li>
</ul>

<p>
We are your trusted partner in navigating the emerging trends and technologies in your domain.
</p>
</div>
</div>
</div>
<div id="outline-container-org03bbd20" class="outline-2">
<h2 id="org03bbd20">Domains</h2>
<div class="outline-text-2" id="text-org03bbd20">
<a href="#top">Back to top</a>

<div id="org2c361da" class="figure">
<p><img src="images/2024-09-08_13-46-48_tubes.jpg" alt="2024-09-08_13-46-48_tubes.jpg" />
</p>
</div>

<p>
The following are brief descriptions of the domains in which we have deep expertise, and through which we provide our services.
</p>
</div>
<div id="outline-container-orgda1d203" class="outline-3">
<h3 id="orgda1d203">High-dimensional flow cytometry</h3>
<div class="outline-text-3" id="text-orgda1d203">
</div>
<div id="outline-container-orgbeff989" class="outline-4">
<h4 id="orgbeff989">Mass cytometry</h4>
<div class="outline-text-4" id="text-orgbeff989">
<p>
We leverage direct wet-lab and dry-lab experience from the lab of Garry Nolan starting in 2012, when mass cytometry was in its infancy, and the best practices were being developed in real time.
</p>
</div>
</div>
<div id="outline-container-orgd1d9eba" class="outline-4">
<h4 id="orgd1d9eba">Spectral flow cytometry</h4>
<div class="outline-text-4" id="text-orgd1d9eba">
<p>
Leveraging our experience in mass cytometry and traditional fluorescnece flow cytometry, we began working and partnering in spectral in the 2020s.
</p>
</div>
</div>
</div>
<div id="outline-container-orgafd29fb" class="outline-3">
<h3 id="orgafd29fb">Single-cell sequencing</h3>
<div class="outline-text-3" id="text-orgafd29fb">
</div>
<div id="outline-container-orgf77ecb2" class="outline-4">
<h4 id="orgf77ecb2">RNA, ATAC, and related (integration included)</h4>
<div class="outline-text-4" id="text-orgf77ecb2">
<p>
Leveraging established best practices and our "first principles" approach from the early days of mass cytometry, we make sure that we deeply understand every detail of our analysis solutions. Our services here include data integration.
</p>
</div>
</div>
<div id="outline-container-orgc797086" class="outline-4">
<h4 id="orgc797086">CITE-seq and related</h4>
<div class="outline-text-4" id="text-orgc797086">
<p>
Here, we leverage our work in flow and mass cytometry to make sure the antibody side is done right, and looks right, along with our expertise in single-cell sequencing on the transcript side.
</p>
</div>
</div>
</div>
<div id="outline-container-orgc752494" class="outline-3">
<h3 id="orgc752494">Spatial proteomics</h3>
<div class="outline-text-3" id="text-orgc752494">
</div>
<div id="outline-container-org36a637f" class="outline-4">
<h4 id="org36a637f">Mass tag based (MIBI and IMC), and fluorescence based (codex)</h4>
<div class="outline-text-4" id="text-org36a637f">
<p>
We utilize our direct experience from the Nolan lab and collaborators, where both MIBI and CODEX were being developed at the same time.
</p>

<p>
This is how we deliver the same "first principles" approach that we deliver for our suspension-based analyses. We emphasize getting all the painful details correct, like cell segmentation.
</p>
</div>
</div>
</div>
<div id="outline-container-org7eefe02" class="outline-3">
<h3 id="org7eefe02">Spatial transcriptomics</h3>
<div class="outline-text-3" id="text-org7eefe02">
</div>
<div id="outline-container-org56c8e83" class="outline-4">
<h4 id="org56c8e83">Core based (eg. Visium), and cell based (eg. Xenium)</h4>
<div class="outline-text-4" id="text-org56c8e83">
<p>
These methods share principles and methods with spatial proteomic analyis. Again, cell segmentation is critical. There are key differences, and we note that there are a number of different technologies to choose from.
</p>

<p>
We emphasize a general understanding of each one and the respective trade-offs, such that we can help you navigate this field as it develops and give you what you need relevant to your project.
</p>
</div>
</div>
</div>
<div id="outline-container-org57e7b49" class="outline-3">
<h3 id="org57e7b49">Network analysis</h3>
<div class="outline-text-3" id="text-org57e7b49">
</div>
<div id="outline-container-org3dfd9c9" class="outline-4">
<h4 id="org3dfd9c9">Pathway analysis</h4>
<div class="outline-text-4" id="text-org3dfd9c9">
<p>
We determine what pathways are likely active in a given cell subset with specific DEGs. We then go into the pathways themselves and determine what genes within the pathway matter. We emphasize the fundamentals of graph theory in this type of work.
</p>
</div>
</div>
</div>
<div id="outline-container-org9311aae" class="outline-3">
<h3 id="org9311aae">Data management</h3>
<div class="outline-text-3" id="text-org9311aae">
</div>
<div id="outline-container-org9e64e17" class="outline-4">
<h4 id="org9e64e17">Knowledge graphs (eg. Neo4J)</h4>
<div class="outline-text-4" id="text-org9e64e17">
<p>
Projects in this domain typically involve organizations that have multi-modal data (genes, drugs, pathways, PPI, diseases, and so forth). Knowledge graphs can put all of this together to produce novel insights you won't get looking at these data separately.
</p>
</div>
</div>
<div id="outline-container-org9b3c26f" class="outline-4">
<h4 id="org9b3c26f">LLM integration</h4>
<div class="outline-text-4" id="text-org9b3c26f">
<p>
A typical project in this domain includes converting English into a given query language (eg. Cypher for Neo4J), such that a knowledge graph can be easily queried to deliver actionable insights.
</p>
</div>
</div>
</div>
</div>
<div id="outline-container-orgd4894d9" class="outline-2">
<h2 id="orgd4894d9">What to expect from an engagement with Burns LSC</h2>
<div class="outline-text-2" id="text-orgd4894d9">
<a href="#top">Back to top</a>

<div id="orgc5f60ae" class="figure">
<p><img src="images/2024-09-08_15-25-11_handshake.jpg" alt="2024-09-08_15-25-11_handshake.jpg" />
</p>
</div>

<p>
A typical engagement begins with a mutual nondisclosure agreement, so we can both talk freely. The project can either be fixed (eg. a specific thing needs to be analyzed by a particular time), or open-ended (eg. continuous advisory). When the details, scope, and deliverables are decided upon, then a statement of work is drafted (sometimes from Burns, sometimes from the client) and signed.
</p>

<p>
The project proceeds with a minimum of one video call per week, so we can always be sync'd on progress toward milestones, and any new developments on both sides. Any amendments to the contract after it has been signed can be done in writing.
</p>

<p>
While most of my clients are remote and most interaction is via email, slack, and video calls, I do prefer to have one or more physical site visits per year. In our experience, this makes a big difference in consultant-client relationships and is critical for longer term projects.
</p>
</div>
</div>
<div id="outline-container-org55c671f" class="outline-2">
<h2 id="org55c671f">What people are saying</h2>
<div class="outline-text-2" id="text-org55c671f">
<a href="#top">Back to top</a>
<p>
<i>“BLS Consulting has brought tremendous value to Alkahest’s research efforts. Their extensive expertise helped us advance cutting-edge projects both in the scientific foundation, as well as the technical implementation. They’re the rare organization that can distill the ambiguity of advanced research down to pragmatic solutions with demonstrable value.”</i>
</p>

<p>
— Scott Lohr, Senior Director of Informatics &amp; Engineering, Alkahest Inc.
</p>

<p>
<i>"Tyler’s expertise in data science, deep biological knowledge, and mastery of cutting-edge technologies have been instrumental in advancing several of our projects. His innovative thinking, coupled with his organization and responsiveness, make him an invaluable consultant. Tyler is our go-to person whenever we encounter challenges, and I greatly appreciate his commitment to efficiency and passion for both the team and the science."</i>
</p>

<p>
— Bahareh Ajami, Assistant Professor of Immunology, Oregon Health &amp; Science University
</p>

<p>
<i>“Tyler’s sharp data science expertise on top of his deep biological background and first principles mindset has given us quality solutions every time we’ve brought him in.”</i>
</p>

<p>
— Xiaoyan Qian, Senior Computational Biologist, 10x Genomics
</p>

<p>
<i>“Tyler’s innovative data analysis solutions has transformed our raw data into actionable insights, enabling us to make smarter and more informed decisions going forward. His fast and engaged approach makes working with him very enjoyable and informative, we look forward to engaging his services again in future projects!”</i>
</p>

<p>
— Natalia Becares, Principal Scientist, LIfT Biosciecnes
</p>
</div>
</div>
<div id="outline-container-org7546634" class="outline-2">
<h2 id="org7546634">Contact us</h2>
<div class="outline-text-2" id="text-org7546634">
<a href="#top">Back to top</a>
</div>
<div id="outline-container-org618ac8c" class="outline-3">
<h3 id="org618ac8c"><a href="https://calendly.com/burnslsc-info/30min">Book a free 30min consultation</a></h3>
<div class="outline-text-3" id="text-org618ac8c">
<p>
If you'd like my two cents on anything, and/or you would like to explore potential engagements/partnerships, you can book a call accordingly.
</p>
</div>
</div>
<div id="outline-container-orga02cca4" class="outline-3">
<h3 id="orga02cca4"><a href="https://www.linkedin.com/in/tylerjburns/">Tyler's LinkedIn</a></h3>
<div class="outline-text-3" id="text-orga02cca4">
<p>
Tyler maintains a strong LinkedIn presence. If you're not connected, simply send a connection request with a message. If he can't answer your questions, he can link you to people who can.
</p>
</div>
</div>
</div>
<div id="outline-container-orgcc5f50d" class="outline-2">
<h2 id="orgcc5f50d">Insights</h2>
<div class="outline-text-2" id="text-orgcc5f50d">
<a href="#top">Back to top</a>
</div>
<div id="outline-container-org70ca0f5" class="outline-3">
<h3 id="org70ca0f5"><a href="https://tjburns08.github.io/social_media_posts.html">A compendium of Tyler's social media posts</a></h3>
<div class="outline-text-3" id="text-org70ca0f5">
<p>
These are grouped by subject, and easier to go through here than trying to browse through LinkedIn history.
</p>
</div>
</div>
<div id="outline-container-orga9eff96" class="outline-3">
<h3 id="orga9eff96"><a href="https://tjburns08.github.io/index.html#single_cell">Single-cell relevant posts</a></h3>
<div class="outline-text-3" id="text-orga9eff96">
<p>
These are markdowns and articles related to all things single-cell, from Tyler's personal website.
</p>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="creator"><a href="https://www.gnu.org/software/emacs/">Emacs</a> 28.1 (<a href="https://orgmode.org">Org</a> mode 9.5.2)</p>
</div>
</body>
</html>
