<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>Tyler Burns's Website</title>
<meta name="generator" content="Org mode" />
<link rel="stylesheet" href="https://cdn.simplecss.org/simple.min.css" />
</head>
<body>
<div id="content">
<h1 class="title">Tyler Burns's Website</h1>


<div id="outline-container-orge49abc6" class="outline-2">
<h2 id="orge49abc6">About me</h2>
<div class="outline-text-2" id="text-orge49abc6">
<p>
I am an American CEO and computational biologist who lives in Berlin, Germany. I am best known for my work on knowledge graphs, and my work on single-cell analysis with respect to the many applications of K-Nearest Neighbors. 
</p>

<p>
I run a company called <a href="https://burnslsc.com/">Burns Life Sciences Consulting, GmbH</a>, which allows me to do science on my terms, work on diverse projects with many organizations, and develop and sell products. The problem is most of my work is behind NDAs so I can't talk about it. There is another large amount of my work that is unpublished, simply because being a CEO has a lot of demands that don't give me the time.
</p>

<p>
This website is a place where I can openly share my ideas and non-client work, regardless of how "finished" it is. I have uploaded written work, code, posters and powerpoint slides of various project I have driven or contributed to.
</p>

<p>
If you have quesitons or comments, or you're interested in working or collaborating with me, just send me a message (see contact info below). I'm always happy to help. 
</p>
</div>
</div>

<div id="outline-container-org1451986" class="outline-2">
<h2 id="org1451986">Articles</h2>
<div class="outline-text-2" id="text-org1451986">
</div>
<div id="outline-container-orgd261e80" class="outline-3">
<h3 id="orgd261e80"><a href="./gpt3_student.html">GPT-3 simulating students</a></h3>
<div class="outline-text-3" id="text-orgd261e80">
<p>
This is a report I wrote for my uncle, who is a professor at the University of Michigan Ross School of Business. The concern was that generative language models would be able to simulate student's responses to essay questions good enough that tech-savvy students would simply offload their homework to GPT-3. I explore this option using my early access to GPT-3, with a conclusion heavily inspired by an article by <a href="https://www.gwern.net/GPT-3#weaknesses">Gwern</a>. You pretty much have to at this point. 
</p>
</div>
</div>

<div id="outline-container-org3daed51" class="outline-3">
<h3 id="org3daed51"><a href="./xai_and_us.html">Explainable AI and humanity</a></h3>
<div class="outline-text-3" id="text-org3daed51">
<p>
I make the connection between understanding a black box AI algorithm (a hot topic) and understanding ourselves. I make the argument that accordingly, we as humans are prepared to take on this task. 
</p>
</div>
</div>

<div id="outline-container-org508c212" class="outline-3">
<h3 id="org508c212"><a href="./just_paint.html">Just paint</a></h3>
<div class="outline-text-3" id="text-org508c212">
<p>
An anecdote from my aunt. The general premise is that while quality is important in many settings, at the end of the day if you're going to paint, you have to paint. I remind myself of this daily in many different contexts.
</p>
</div>
</div>

<div id="outline-container-org2c32a0c" class="outline-3">
<h3 id="org2c32a0c"><a href="https://medium.com/@tjburns_72591/how-to-utilize-scientific-literature-trends-to-gain-intuition-about-a-topic-b5c554e3d280">How to utilize scientific literature trends to gain intuition around a topic</a></h3>
<div class="outline-text-3" id="text-org2c32a0c">
<p>
The scientific literature is overwhelming, and knowing how to utilize text mining and analytic tools can help you efficiently get what you want out of a literature search. Here, I utilize the PubMed API to find publication rates for particular topics. I show that among other things, single-cell sequencing began out-pacing mass cytometry in 2016. Insight like this helps you see how crowded a field is and especially identify trends.
</p>
</div>
</div>

<div id="outline-container-org2b3c070" class="outline-3">
<h3 id="org2b3c070"><a href="https://medium.com/coinmonks/how-to-identify-thought-leaders-and-visualize-their-influence-c01aa218090e">How to identify thought leaders and visualize their influence</a></h3>
<div class="outline-text-3" id="text-org2b3c070">
<p>
Understanding how authors of a given field are connected can help you identify key individuals to pay attention to. Here is how I utilize the PubMed API to build co-author networks, which lead me to identify thought leaders in a given domain. In this article, focusing on mass cytometry, I identify two types of thought leaders: one exclusive to a particular sub-domain, and one who spans across multiple sub-domains. It is important to know both types when approaching a new topic.
</p>
</div>
</div>

<div id="outline-container-orgbd3579f" class="outline-3">
<h3 id="orgbd3579f"><a href="https://medium.com/@tjburns_72591/using-and-mining-pre-prints-to-stay-ahead-of-your-field-with-the-help-of-twitter-50d5bdc528de">Using and mining pre-prints to stay ahead of your field, with the help of Twitter</a></h3>
<div class="outline-text-3" id="text-orgbd3579f">
<p>
I explain why pre-prints are important to staying ahead of the technology and general paradigms in your field, with single-cell analysis as an example. I then show how I utilize the Twitter API to harvest and rank tweets from automated pre-print linking bots from bioRxiv to determine what pre-print articles are being talked about (and therefore what you should probably pay attention to).
</p>
</div>
</div>
</div>


<div id="outline-container-orge30e2e0" class="outline-2">
<h2 id="orge30e2e0">First author publications</h2>
<div class="outline-text-2" id="text-orge30e2e0">
</div>
<div id="outline-container-org3d4f4ab" class="outline-3">
<h3 id="org3d4f4ab"><a href="https://www.biorxiv.org/content/10.1101/337485v1">Continuous Visualization of Multiple Biological Conditions In Single-Cell Data</a></h3>
<div class="outline-text-3" id="text-org3d4f4ab">
<p>
Abstract: In high-dimensional single cell data, comparing changes in functional markers between conditions is typically done across manual or algorithm-derived partitions based on population-defining markers. Visualizations of these partitions is commonly done on low-dimensional embeddings (eg. t-SNE), colored by per-partition changes. Here, we provide an analysis and visualization tool that performs these comparisons across overlapping k-nearest neighbor (KNN) groupings. This allows one to color low-dimensional embeddings by marker changes without hard boundaries imposed by partitioning. We devised an objective optimization of k based on minimizing functional marker KNN imputation error. Proof-of-concept work visualized the exact location of an IL-7 responsive subset in a B cell developmental trajectory on a t-SNE map independent of clustering. Per-condition cell frequency analysis revealed that KNN is sensitive to detecting artifacts due to marker shift, and therefore can also be valuable in a quality control pipeline. Overall, we found that KNN groupings lead to useful multiple condition visualizations and efficiently extract a large amount of information from mass cytometry data. Our software is publicly available through the Bioconductor package Sconify.
</p>
</div>
</div>

<div id="outline-container-orgbf0f7ec" class="outline-3">
<h3 id="orgbf0f7ec"><a href="https://pubmed.ncbi.nlm.nih.gov/28094900/">High Throughput Precision Measurement of Nuclear Localization in Subcellular Localization in Single Cells</a></h3>
<div class="outline-text-3" id="text-orgbf0f7ec">
<p>
Abstract: To quantify visual and spatial information in single cells with a throughput of thousands of cells per second, we developed Subcellular Localization Assay (SLA). This adaptation of Proximity Ligation Assay expands the capabilities of flow cytometry to include data relating to localization of proteins to and within organelles. We used SLA to detect the nuclear import of transcription factors across cell subsets in complex samples. We further measured intranuclear re-localization of target proteins across the cell cycle and upon DNA damage induction. SLA combines multiple single-cell methods to bring about a new dimension of inquiry and analysis in complex cell populations. Â© 2017 International Society for Advancement of Cytometry.
</p>

<p>
My summer students are co-authors on this paper! Undergrads and high school students. They worked very hard and learned a lot. I am proud of each and every one of them. 
</p>
</div>
</div>

<div id="outline-container-org2d97649" class="outline-3">
<h3 id="org2d97649"><a href="./Burns.Dissertation.Final.pdf">Expanding the Capabilities of Mass Cytometry Data Acquisition and Analysis</a></h3>
<div class="outline-text-3" id="text-org2d97649">
<p>
My PhD thesis dissertation, from the laboratory of Garry P. Nolan at Stanford University School of Medicine. 
</p>

<p>
In sum: I started by developing a method to enable flow and mass cytometry to detect and quantify nuclear localization, called Subcellular Localization Assay (SLA), which came out of a collaboration with the lab of Ola Soederberg at University of Uppsala, Sweden.
</p>

<p>
In parallel, I was taking computer science classes as a side hobby. I reached a point where I was trying to compare two t-SNE maps between unstimulated and simulated data, and I realized that there was a K-Nearest Neighbors based solution that I could implement with my newfoud computer science competencies. I therefore developed Sconify, a now BioConductor package that allows for these visualizations. There were many use cases, and I spent the remainder of my thesis developing this method further and doing various collaborations with it. 
</p>
</div>
</div>
</div>

<div id="outline-container-orgfdc8e41" class="outline-2">
<h2 id="orgfdc8e41">Talks</h2>
<div class="outline-text-2" id="text-orgfdc8e41">
</div>
<div id="outline-container-org347ddbe" class="outline-3">
<h3 id="org347ddbe"><a href="./tjb_dimr_talk.pdf">A visual interrogation of dimension reduction tools for single-cell analysis</a></h3>
<div class="outline-text-3" id="text-org347ddbe">
<p>
German CyTOF User Forum; Berlin, Germany; January 2020.
In this talk, I measured the accurracy of dimension reduction tools (PCA, t-SNE, and UMAP) in terms of their nearest neighbor overlap. This is the k-nearest neighbors of a given cell in the original high dimension space, in comparison to the k-nearest neighbors of a given cell in the embedding. I show that the overlap here is much lower than my audience expected. I've given this talk many times since then, for my clients.
</p>
</div>
</div>

<div id="outline-container-org42fe446" class="outline-3">
<h3 id="org42fe446"><a href="./visual_capabilities_of_som.pdf">Neighborhood-based analysis of self-organizing maps</a></h3>
<div class="outline-text-3" id="text-org42fe446">
<p>
<a href="https://vib.be/labs/saeys-lab">Laboratory of Yvan Saeys</a>, VIB Ghent, Belgium. June 2018.
This slide deck summarizes some work I did with Sofie Van Gassen, developer of <a href="https://bioconductor.org/packages/release/bioc/html/FlowSOM.html">FlowSOM</a> and all-around awesome person. We were looking at what is called the U-Matrix, a way to visualize the self organizing maps that FlowSOM produces. The question was what insights could we derive from using the U-Matrix to visualize the output of very large FlowSOM clusterings (eg. a 100 x 100 grid rather than the default 10 x 10). So far as I know, this is not explored in any major CyTOF publication, so any CyTOF users who use FlowSOM (most people at the time of writing) should have a look at this. There are visualizations in here that are useful but remain unpublished.
</p>
</div>
</div>

<div id="outline-container-orgfe45132" class="outline-3">
<h3 id="orgfe45132"><a href="./mass.cytometry.analysis.history.pdf">A history of mass cytometry data analysis, and where the field is going</a></h3>
<div class="outline-text-3" id="text-orgfe45132">
<p>
<a href="https://www.drfz.de/en/aktuelles/veranstaltungen/cytof-forum-2020/">German Rheumatism Research Center</a>; Berlin, Germany; March 2019.
I talk about how CyTOF data analysis developed from its inception at the beginning of 2010 to now. In doing so, I provide a template for proper CyTOF data analysis in terms of how we got there. In doing so, I test various assumptions: I show visualizations of data transformations other than asinh(x/5), and I show what a SPADE tree looks like with completely random inputs. I like to show these slides to people new to CyTOF data analysis to properly orient them. 
</p>
</div>
</div>

<div id="outline-container-org3210d5c" class="outline-3">
<h3 id="org3210d5c"><a href="./drfz_tsne_interrogation_talk_final.pdf">A comprehensive interrogation of the t-SNE algorithm for mass cytometry analysis</a></h3>
<div class="outline-text-3" id="text-org3210d5c">
<p>
German Rheumatism Research Center; Berlin, Germany; May 2018.
This talk was a response to a member of the research institue who was simply not convinced that t-SNE was providing the accurracy that the avearge CyTOF user thought. In this talk, I show that he was right. This being said, I provide recommendations for how to properly use t-SNE for CyTOF analysis.
</p>
</div>
</div>

<div id="outline-container-orgb19d4dd" class="outline-3">
<h3 id="orgb19d4dd"><a href="./burns_cytof_user_forum_talk_for_pdf.pdf">Nearest neighborhood comparisons across biological conditions in single cell data</a></h3>
<div class="outline-text-3" id="text-orgb19d4dd">
<p>
Invited Speaker, German CyTOF User Forum; Berlin, Germany; February 2018.
This is the talk version of my 2018 Sconify paper, that ended up being the final chapter of my PhD thesis. There are two aspects to this talk. The first is making visual comparisons of unstimulated and stimulated CyTOF data when looking at measurements of phosphoproteins. This was easily done on SPADE trees, but not t-SNE maps, until I started making k-nearest neighbor based comparisons. The second aspect of this talk is using the same nearest neighbor based comparisons to investigate batch effects in CyTOF data. I note that batch effects were only heavily discussed among CyTOF users starting near 2020 (in my circles), and this work goes back to 2016.
</p>
</div>
</div>
</div>

<div id="outline-container-org9332d5b" class="outline-2">
<h2 id="org9332d5b">Posters</h2>
<div class="outline-text-2" id="text-org9332d5b">
</div>
<div id="outline-container-orgfb4c833" class="outline-3">
<h3 id="orgfb4c833"><a href="./final_distance.project.poster.pdf">Determining which distance metrics are ideal within a mass cytometry data analysis pipeline</a></h3>
<div class="outline-text-3" id="text-orgfb4c833">
<p>
CYTO Conference; Prague, Czech Republic; May 2018.
Abstract: Due to the rise of high-dimensional single cell technologies in the past few years, there has been an increasing number of both computational methods and workflows to analyze the new wealth of data. However, non-intuitive properties of high-dimensional space can give rise to analysis artifacts, collectively known of as the âcurse of dimensionality.â Increasing dimensions differentially affect the performance of distance metrics, and there is no clear consensus about which distance metrics to use for which analysis strategies. While the influence of many tool-specific parameters has been evaluated, we study here the impact of commonly used distance metrics on the outcome of dimensionality reduction and clustering.
</p>
</div>
</div>
</div>

<div id="outline-container-orgb9d7cdc" class="outline-2">
<h2 id="orgb9d7cdc">Software</h2>
<div class="outline-text-2" id="text-orgb9d7cdc">
</div>
<div id="outline-container-org3d8168b" class="outline-3">
<h3 id="org3d8168b"><a href="https://www.bioconductor.org/packages/release/bioc/html/Sconify.html">Bioconductor package Sconify</a></h3>
<div class="outline-text-3" id="text-org3d8168b">
<p>
Official description: This package does k-nearest neighbor based statistics and visualizations with flow and mass cytometery data. This gives tSNE maps"fold change" functionality and provides a data quality metric by assessing manifold overlap between fcs files expected to be the same. Other applications using this package include imputation, marker redundancy, and testing the relative information loss of lower dimension embeddings compared to the original manifold.
</p>
</div>
</div>
</div>

<div id="outline-container-orga468d22" class="outline-2">
<h2 id="orga468d22">Markdowns</h2>
<div class="outline-text-2" id="text-orga468d22">
<p>
These are primarily R and python markdowns exploring various topics. I apologize if the one you're reading is a bit messy. Most of these are instances of me just tinkering for the sake of tinkering. These often lead to interesting results that I don't have time to follow up on, but I figure that showing unpolished work is better than showing no work. If you have any questions or comments, please feel free to contact me. Anything that gets attention here will trigger me to work on it.
</p>

<p>
There are two potenital audiences here. The first are those with questions about any of the topics below that I cover. The second are those learning coding or data science who want to see examples of simple but insightful analyses.
</p>

<p>
Regardless of the reader's intent, I hope above all that the work below will encourage people to really tinker with text, numbers, and data. You can get surprisingly far if you get in the habit of it. 
</p>
</div>


<div id="outline-container-org260698d" class="outline-3">
<h3 id="org260698d"><a href="./cytof_data_transformations.html">Data transformations for CyTOF</a></h3>
<div class="outline-text-3" id="text-org260698d">
<p>
CyTOF data are transformed using the inverse hyperbolic sine (asinh) of the data divided by 5 (aka scale argument of 5). But does it have to be like that? What happens if we use a scale argument of 1? 500? What if we do a log transform? How does t-SNE look on untransformed CyTOF data?
</p>
</div>
</div>

<div id="outline-container-org418c880" class="outline-3">
<h3 id="org418c880"><a href="./1_law_large_numbers_central_limit_theorem.html">Coin toss series 1: The law of large numbers and the central limit theorem</a></h3>
<div class="outline-text-3" id="text-org418c880">
<p>
I taught one of my high school summer students the basics of probability by simulating coin tosses in R. Here, we "discover" the law of large numbers and the central limit theorem using simulated coin tosses. 
</p>
</div>
</div>

<div id="outline-container-orgbd3ac8e" class="outline-3">
<h3 id="orgbd3ac8e"><a href="./2_runs_of_luck.html">Coin toss series 2: Runs of luck</a></h3>
<div class="outline-text-3" id="text-orgbd3ac8e">
<p>
Here, we build on the initial piece in the series by looking at the properties of runs of luck. If we flip a coin a million times, how often will we get 10 heads in a row? How many times do we need to flip a coin to get 20 heads in a row on average? Related to sports. How often, statistically, would you expect Steph Curry to make 10 three pointers in a row given his 3-point shot percentage? 
</p>
</div>
</div>

<div id="outline-container-orgc2d5661" class="outline-3">
<h3 id="orgc2d5661"><a href="./3_fair_vs_unfair_coins.html">Coin toss series 3: Fair versus unfair coins</a></h3>
<div class="outline-text-3" id="text-orgc2d5661">
<p>
Here, we examine the properties of unfair coins, where the odds of getting heads or tails does not equal 50%. Can we figure out whether a coin is a fair coin? 
</p>
</div>
</div>

<div id="outline-container-org983ecd0" class="outline-3">
<h3 id="org983ecd0"><a href="./4_random_walk.html">Coin toss series 4: Random walks</a></h3>
<div class="outline-text-3" id="text-org983ecd0">
<p>
Here, we show that if we simulate flipping coins, but we keep a record of the number of heads and the number of tails, we end up doing a random walk. We visualize these walks (they look somewhat like stock market data), and ask questions like how often a random walker crosses zero.  
</p>
</div>
</div>

<div id="outline-container-org756fe1b" class="outline-3">
<h3 id="org756fe1b"><a href="./5_dice_roll.html">Coin toss series 5: Dice rolls</a></h3>
<div class="outline-text-3" id="text-org756fe1b">
<p>
Here, we do an abstraction of the coin tosses we have been simulating, by coinsidering dice of three or more faces. We simulate these dice rolls and examine their properties. How often does a six sided dice land on the number 3? We can figure that out with simple math, but if you roll a dice 1000 times, and you do that again, and you do that again, what will be the standard deviation of the number of times the dice lands on 3? 
</p>
</div>
</div>

<div id="outline-container-orga846ba0" class="outline-3">
<h3 id="orga846ba0"><a href="./6_is_this_sequence_random.html">Coin toss series 6: Is this sequence random?</a></h3>
<div class="outline-text-3" id="text-orga846ba0">
<p>
Here, we look at fair coin tosses, unfair coin tosses, and random walks, and explore the randomness of the sequences by doing convolutions on the sequences with kernel size 2.
</p>
</div>
</div>

<div id="outline-container-org8cd52d8" class="outline-3">
<h3 id="org8cd52d8"><a href="./1d_ca.html">1-D Cellular Automata</a></h3>
<div class="outline-text-3" id="text-org8cd52d8">
<p>
Here, I write some code to produce each of the 256 Wolfram cellular automata rules, and visualize the output. 
</p>
</div>
</div>

<div id="outline-container-org6d49de9" class="outline-3">
<h3 id="org6d49de9"><a href="./explore_rule_110.html">Explore Wolfram Rule 110</a></h3>
<div class="outline-text-3" id="text-org6d49de9">
<p>
Here, I write some code to produce Rule 110, a Class 4 1-D cellular automata. I then enhance the gliders to make them easier to see. I explore how the output changes if I make the rule probabilistic (eg. 99.99% chance the rule will be followed. 
</p>
</div>
</div>

<div id="outline-container-orge6dd1d8" class="outline-3">
<h3 id="orge6dd1d8"><a href="./question_graph_writeup.html">Question graph</a></h3>
<div class="outline-text-3" id="text-orge6dd1d8">
<p>
You are only as good as the questions you ask yourself and others. My uncle told me that many years ago when I was getting started with my career and it stuck. This has been relevant to me in terms of having and maintaining good friendships, being a good husband, being a good family member, being a good businessman, and when I was in graduate school, being a good scientist, and simply being an interesting person. I have a very large list of questions now that is very overwhelming. So I turned them into an embedding using the BERT language model, turned that into a nearest neighbor graph, and then derived insight from looking at the questions in terms of "communities." 
</p>
</div>
</div>
</div>


<div id="outline-container-orgc26194f" class="outline-2">
<h2 id="orgc26194f">Other contributions</h2>
<div class="outline-text-2" id="text-orgc26194f">
</div>
<ul class="org-ul">
<li><a id="org526a550"></a>Former computational biologist, now guest researcher, at the German Rheumatism Research Center in Berlin, Germany. I will always stay connected to my academic roots.<br /></li>
<li><a id="org95a3d2a"></a>Developing software to interrogate and visualize the local similarities between original manifolds and lower dimensional embeddings. Important for anyone wanting to determine which of these methods is the right tool for the job.<br /></li>
<li><a id="orgdc76632"></a>Solved a long-standing data visualization problem for mass cytometry, and developed a Bioconductor package for it, with a visual description here, and this publication..<br /></li>
<li><a id="org5f334d7"></a>Guest blog contributor for Cytobank, Inc. focusing on how to properly use t-SNE for mass cytometry analysis.<br /></li>
<li><a id="org55892b3"></a>Set the best practices in my PhD thesis lab for learning bioinformatics as a classical biologist.<br /></li>
<li><a id="orgbe86d63"></a>Helped develop a wet-lab implementation for a cutting-edge bioinformatics concept, which became part of a patent.<br /></li>
<li><a id="orgab1a211"></a>Led an international collaboration between my thesis lab (USA) and a laboratory in Uppsala, Sweden, whose biochemical foundations turned into this publication, and helped this one.<br /></li>
<li><a id="org48a9046"></a>Mentored several high school students and undergraduates throughout my thesis work, teaching them biology, computer science, statistics, and importantly how to integrate these fields.<br /></li>
<li><a id="org1b3fbe9"></a>Built a website for my PhD program using HTML, JavaScript, and SQL, that helped first year students connect with current and previous members of a given research laboratory (I'd link it but you have to be a Stanford student to view it).<br /></li>
</ul>
</div>


<div id="outline-container-org028f954" class="outline-2">
<h2 id="org028f954">Links and contact info</h2>
<div class="outline-text-2" id="text-org028f954">
</div>
<div id="outline-container-org68d7f6a" class="outline-4">
<h4 id="org68d7f6a"><a href="https://www.linkedin.com/in/tylerjburns/">LinkedIn</a></h4>
</div>
<div id="outline-container-org66bc5f9" class="outline-4">
<h4 id="org66bc5f9"><a href="https://twitter.com/tjburns08">Twitter</a></h4>
</div>
<div id="outline-container-org19fa582" class="outline-4">
<h4 id="org19fa582"><a href="https://github.com/tjburns08">GitHub</a></h4>
</div>
<div id="outline-container-orgfd7a56e" class="outline-4">
<h4 id="orgfd7a56e"><a href="https://burnslsc.com/">Company Website</a></h4>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="creator"><a href="https://www.gnu.org/software/emacs/">Emacs</a> 26.3 (<a href="https://orgmode.org">Org</a> mode 9.1.9)</p>
<p class="validation"></p>
</div>
</body>
</html>
